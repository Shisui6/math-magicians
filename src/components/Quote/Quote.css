.quote {
  height: calc(100vh - 72px);
  background-image: url('../../../public/home.jpg');
  background-size: 100% 100%;
  background-position: center;
  background-repeat: no-repeat;
  opacity: 0;
  animation: fade-in 0.8s forwards;
}

@keyframes fade-in {
  100% {
    opacity: 1;
  }
}

.quote p {
  text-align: center;
  width: 80%;
  margin: 0 auto;
  padding-top: 170px;
  font-size: 25px;
  font-weight: 500;
  line-height: 45px;
  color: #073066;
}

@media screen and (max-width: 768px) {
  .quote {
    height: calc(100vh - 53px);
  }
}
