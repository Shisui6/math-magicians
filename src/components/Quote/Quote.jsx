import './Quote.css';

const Quote = () => {
  const quotes = ['Pure mathematics is, in its way, the poetry of logical ideas. - Albert Einstein', 'Mathematics is the most beautiful and most powerful creation of the human spirit. - Stefan Banach', 'Mathematics is the music of reason. - James Joseph Sylvester', 'Mathematics knows no races or geographic boundaries; for mathematics, the cultural world is one country. - David Hilbert', 'It is impossible to be a mathematician without being a poet in soul. - Sofia Kovalevskaya', 'A mathematician who is not also something of a poet will never be a complete mathematician. - Karl Weierstrass', 'Mathematics compares the most diverse phenomena and discovers the secret analogies that unite them. - Joseph Fourier', 'Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. - William Paul Thurston', 'In mathematics the art of proposing a question must be held of higher value than solving it. - Georg Cantor', 'Go down deep enough into anything and you will find mathematics. - Dean Schlicter'];

  return (
    <div className="quote">
      <p data-testid="5">{quotes[0]}</p>
    </div>
  );
};

export default Quote;
