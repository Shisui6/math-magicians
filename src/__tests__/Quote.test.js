import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Quote from '../components/Quote/Quote';

describe('Quote', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Quote />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('contains the p element', () => {
    render(<Quote />);

    expect(screen.getByTestId('5')).toHaveTextContent('Pure mathematics is, in its way, the poetry of logical ideas. - Albert Einstein');
  });
});
