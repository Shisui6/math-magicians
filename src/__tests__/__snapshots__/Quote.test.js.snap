// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote renders correctly 1`] = `
<div
  className="quote"
>
  <p
    data-testid="5"
  >
    Pure mathematics is, in its way, the poetry of logical ideas. - Albert Einstein
  </p>
</div>
`;
